cmd/influx_inspect/buildtsi/util.go

```diff
@@ -0,0 +1,218 @@
+package buildtsi
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/influxdata/influxdb/models"
+	"github.com/influxdata/influxdb/tsdb"
+	"github.com/influxdata/influxdb/tsdb/engine/tsm1"
+	"github.com/influxdata/influxdb/tsdb/index/tsi1"
+	"go.uber.org/zap"
+)
+
+func IndexShard(sfile *tsdb.SeriesFile, dataDir, walDir string, maxLogFileSize int64, batchSize int, log *zap.Logger, verboseLogging bool) error {
+	log.Info("Rebuilding shard")
+
+	// Check if shard already has a TSI index.
+	indexPath := filepath.Join(dataDir, "index")
+	log.Info("Checking index path", zap.String("path", indexPath))
+	if _, err := os.Stat(indexPath); !os.IsNotExist(err) {
+		log.Info("tsi1 index already exists, skipping", zap.String("path", indexPath))
+		return nil
+	}
+
+	log.Info("Opening shard")
+
+	// Find shard files.
+	tsmPaths, err := collectTSMFiles(dataDir)
+	if err != nil {
+		return err
+	}
+	walPaths, err := collectWALFiles(walDir)
```

Comment:
Have you tested what happens when we run this in an environment without a WAL? I think `collectWALFiles` would return an error, which means rebuilding will fail. If that's the case we can either treat `walDir == ""` as a special case in `collectWALFiles` or we can check here, and only call `collectWALFiles` in the case `walDir != ""`.